framework/src/packets/ip/v6/srh.rs

```diff
@@ -134,7 +134,21 @@ pub struct BadSegmentsError;
     display = "Segments left {} must be less than segment list length {}",
     _0, _1
 )]
-pub struct SegmentsLeftOutOfBoundError(u8, usize);
+pub struct SegmentsLeftOutOfBoundsError(u8, usize);
+
+/// Error for bad index within segment list
+#[derive(Debug, Fail)]
+#[fail(display = "Index {} out of bounds of segment list length {}", _0, _1)]
+pub struct SegmentsIndexOutOfBoundsError(usize, usize);
```

Comment:
should these two errors be combined? they look very similar.

Reply:
Sure